sync/src/main/java/tech/pegasys/teku/sync/PeerSync.java

```diff
@@ -119,18 +128,29 @@ public void stop() {
               final SafeFuture<Void> readyForNextRequest =
                   asyncRunner.getDelayedFuture(
                       NEXT_REQUEST_TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
-              return peer.requestBlocksByRange(startSlot, count, STEP, this::blockResponseListener)
-                  .thenApply((res) -> readyForNextRequest);
+              final PeerSyncBlockRequest request =
+                  new PeerSyncBlockRequest(
+                      readyForNextRequest, startSlot.plus(count), this::blockResponseListener);
+              return peer.requestBlocksByRange(startSlot, count, STEP, request)
+                  .thenApply((res) -> request);
             })
         .thenCompose(
-            (readyForNextRequest) -> {
+            (blockRequest) -> {
+              final UnsignedLong nextSlot = blockRequest.getActualEndSlot().plus(UnsignedLong.ONE);
               LOG.trace(
-                  "Completed request for {} blocks starting at {} from peer {}",
+                  "Completed request for {} blocks starting at {} from peer {}. Next request starts from {}",
```

Comment:
isn't this more like `Completed request for {} blocks starting at {} from peer {}, got {} blocks. Next request starts from {}`?

Reply:
We don't actually count the number of blocks received - we just track what the last slot we got up to was. Reworded to talk about slots since that's technically what the range specifies.